crates/sui-indexer/src/handlers/committer.rs

```diff
@@ -82,11 +83,24 @@ where
                     );
                 })?;
             }
+            // stop adding to the commit batch if we've reached the end checkpoint
+            if let Some(end_checkpoint_sequence_number) = end_checkpoint_opt {
+                if next_checkpoint_sequence_number > end_checkpoint_sequence_number {
+                    break;
+                }
+            }
         }
         if !batch.is_empty() {
             commit_checkpoints(&state, batch, None, &metrics, &mut committed_checkpoints_tx).await;
             batch = vec![];
         }
+
+        // stop the commit task if we've reached the end checkpoint
+        if let Some(end_checkpoint_sequence_number) = end_checkpoint_opt {
+            if next_checkpoint_sequence_number > end_checkpoint_sequence_number {
+                break;
+            }
+        }
```

Comment:
anticipated q: break here does not stop the binary entirely, the ingestion framework part is still running, which is fine for benchmark.

Reply:
I suppose after a while, the ingestion part will get stuck because the receiver does not receive checkpoints anymore? Which is indeed fine for benchmarking

Reply:
ya I would like to extend the change and shut down the whole binary if we decide to continue the prod codes here, but might not worth it if we go with indexer-alt.